src/inout.f90

```diff
@@ -165,90 +200,176 @@ END SUBROUTINE Input
+  USE MyNumbers
+  USE IPara
+  USE DPara
 
-  INTEGER,INTENT(in)         :: n
-  REAL(8),INTENT(in)         :: vdata(n)
-  CHARACTER(len=*),INTENT(in) :: fnamestr
+  IMPLICIT NONE
 
-  CHARACTER(len=*)   :: GetFileName
-  CHARACTER(len=500) :: str,fstr,vstr
+  CHARACTER*200          :: MakeMiddleName
+  INTEGER, INTENT(in)    :: IWidth
+  INTEGER, INTENT(inout) :: IErr
+  CHARACTER*200 middlenamestr, mnstr1, mnstr2, mnstr3
+  CHARACTER*1 SymbolCP,symbolLP
   
-  INTEGER           :: i
+  !PRINT*,"DBG: MakeMiddleName()"
+
+  IErr= 0
+
+  IF(CubeConPot.GE.0.0D0) THEN
+     SymbolCP="+"
+  ELSE
+     SymbolCP="-"
+  END IF
+
+  IF(LiebConPot.GE.0.0D0) THEN
+     SymbolLP="+"
+  ELSE
+     SymbolLP="-"
+  END IF
 
-  str = fnamestr
+  !   WRITE out the input parameter
+  WRITE(mnstr1, &
+   '(A1,I1,I1,A2,I4.4,A3)') &
+       "L", Dim, Nx, &
+       "_M", IWidth, &
+       "_Es"
+  
+  WRITE(mnstr2, &
+   '(A3,A1,I6.6,A3,I6.6,A3,A1,I6.6,A3,I6.6)') &
+       "_CP", SymbolCP,NINT(100.*ABS(CubeConPot)), &
+       "_CD", NINT(100.*ABS(CubeDis)), &
+       "_LP", SymbolLP,NINT(100.*ABS(LiebConPot)), &
+       "_LD", NINT(100.*ABS(LiebDis)) 
   
-  DO i=1,n
-     
-     fstr = str(INDEX(str,'('):INDEX(str,')'))
-     IF(fstr(2:2).EQ.'I')THEN
-        WRITE(vstr,fstr) INT(vdata(i))
-     ELSE
-        WRITE(vstr,fstr) vdata(i)
-     END IF
+  WRITE(mnstr3, &
+   '(A3,I6.6,A3,I6.6)') &
+       "_oS", NINT(100.*ABS(OffDShift)), &
+       "_oD", NINT(100.*ABS(OffDDis))
   
-     str = str(1:INDEX(str,'(')-1) // TRIM(ADJUSTL(vstr)) // str(INDEX(str,')')+1:LEN(str)) 
-     
-  END DO
+!!$  PRINT*, TRIM(mnstr1)
+!!$  PRINT*, TRIM(mnstr2)
+!!$  PRINT*, TRIM(mnstr3)
+  middlenamestr= TRIM(mnstr1) // TRIM(mnstr2) // TRIM(mnstr3)
   
-  GetFileName = TRIM(ADJUSTL(str))
+  MakeMiddleName = TRIM(ADJUSTL(middlenamestr))
 
   RETURN
-END FUNCTION GetFileName
+END FUNCTION MakeMiddleName
 
 !--------------------------------------------------------------------
-! CheckOutput:
+! MakeDataDir:
 !
 ! IErr	error code
 
-SUBROUTINE CheckOutput( Dim, Nx, IWidth, CubeDis, LiebDis, CubeConPot, LiebConPot, PreSeed, str, IErr )
+!!$SUBROUTINE MakeDataDir(Dim, Nx, Width, CubeDis, LiebDis, CubeConPot, LiebConPot, str)
+SUBROUTINE MakeDataDir(IWidth, dirname, IErr)
 
-  USE MyNumbers 
-  USE IChannels
-!  USE DPara
-!  USE IPara
+  USE MyNumbers
+  USE IPara
+  USE DPara
 
+  INTEGER IWidth, IErr
+  CHARACTER(len=*) dirname
+  !CHARACTER*200 MakeDataDir
+
+  LOGICAL*4  ierr1
+  EXTERNAL SYSTEM
   
-  INTEGER(KIND=IKIND) Dim, Nx, IWidth, IErr, PreSeed, ISSeed
-  REAL(KIND=RKIND) CubeDis, LiebDis, CubeConPot, LiebConPot
-  
-  CHARACTER*100 FileName, str
-  CHARACTER*1 SymbolCP,symbolLP
+  PRINT*, "MakeDataDir(): checking for ", TRIM(dirname)
 
-  IF(CubeConPot.GE.0.0D0) Then
-     SymbolCP="+"
-  Else
-     SymbolCP="-"
-  End If
+#ifdef ifort
+  INQUIRE(directory=TRIM(ADJUSTL(dirname)), Exist=ierr1) ! ifort
+#else
+  INQUIRE(file=TRIM(ADJUSTL(dirname)), Exist=ierr1) ! gfortran
```

Comment:
Why for compiler ifort the parameter is "directory" but for gfortran it is "file"

Reply:
Good question, but probably more for the compiler people. This is what seems to work for us at the moment. Need to check if compiler updates may have resolved the issue.